### PR TITLE
Extend visibility timeout when dequeuing test result messages

### PR DESCRIPTION
**Pull request by mehansen**
# DEVOPS PULL REQUEST

## Related Issue

follow up on what was discussed in 10/27 eng sync - increase time a message is 'invisible' (can't be dequeued again) after being dequeued

## Changes Proposed
increase the visibility timeout when grabbing messages from the test results publishing queue. the default is 30 seconds and occasionally we grab (and subsequently report) a message multiple times if the queue wasn't emptied in 30 seconds
a message being invisible doesn't prevent it from being deleted if the pop receipt is still valid

## Additional Information
- more info on visibilitytimeout: https://learn.microsoft.com/en-us/rest/api/storageservices/get-messages#remarks
- RS wants to turn off allowDuplicates (currently true) for all senders. I think it's probably okay for them to do that after this change. or we can turn it off with a request param (overrides RS settings) so we can more easily test in a lower

## Testing

will deploy to a lower when one opens up and make sure the queue still works and messages are deleted even with the longer timeout

### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Comment by mehansen** [comment]
it works!
pushed logging and a 45 second timeout to `dev4` and could see from the logs that the same 2 messages were dequeueing and getting picked up and sent multiple times:
Screenshot 2022-12-02 at 1 07 12 PM
Screenshot 2022-12-02 at 1 10 06 PM

then I merged in this branch and confirmed that messages were only dequeued and sent once despite the 45 second wait:
Screenshot 2022-12-02 at 2 11 43 PM

if interested, you can see my test code on https://github.com/CDCgov/prime-simplereport/tree/merethe/test-dequeue-timeout

File: ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ const {
   AZ_STORAGE_QUEUE_SVC_URL,
 } = ENV;
 const DEQUEUE_BATCH_SIZE = 32;
+const DEQUEUE_TIMEOUT = 60 * 60;  // length of time in seconds a message is invisible after being dequeued
```
Comment (zdeveloper) [inline ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts:28]: so this is changing it to an hour from 30 seconds, is that the intention?

Comment (zdeveloper) [inline ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts:28]: So this is changing it to an hour from 30 seconds, is that the intention?

**Review by zdeveloper — COMMENTED** [review]
Can you elaborate on what the original problem is this solving? I was OOO when the dev discussion took place and would like some more context.

**Comment by mehansen** [comment]
@zdeveloper yes definitely I think a deeper explanation is a good idea, even for other folks since it's been a while since we talked about this

> I am not sure I understand what this statement is claiming, to my knowledge, once a message is picked up, it can not be picked up again unless its placed back on the queue (more accurately allowed to be picked up again once its not deleted). is that a correct statement? How queue being emptied comes in play?

that's actually not true. a message can't be picked up until the [visibility timeout](https://learn.microsoft.com/en-us/rest/api/storageservices/get-messages#remarks) expires, which by default is set to 30 seconds

> How is this change going to prevent duplicates?

when the loop for dequeueing messages takes longer than 30 seconds, we start picking up messages that have already been dequeued in a previous iteration of the loop. we end up sending them to RS multiple times in the same batch. extending the timeout means a message won't be visible on the queue during the same function run that picked it up the first time and so we won't accidentally add it twice to our csv that we send to RS

File: ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts (hunk 1 above)
Comment (mehansen) [reply ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts:28]: yes, 1 hr is a little bit arbitrary but it's a larger number that just is meant to be long enough that our function can delete the message before it reappears again.

**Comment by mehansen** [comment]
re [this comment](https://github.com/CDCgov/prime-simplereport/pull/4661#issuecomment-1335920134) where I describe my testing: I'm confirming the behavior by adding a wait in our dequeue function that causes it to take longer than 30 seconds, and we can see in the logs that the same messages are being picked up multiple times in the same function run. the dequeue function usually doesn't take that long, but sometimes it does, esp if there are many messages on the queue or Azure is being a little slow retrieving the messages

**Review by zdeveloper — APPROVED** [review]
LGTM! thanks for the explanation!

**Review by nathancrtr — APPROVED** [review]
LGTM! Thanks for providing such a clear explanation and proof of testing 🌮

**Pull request merged**